crates/polars-time/src/chunkedarray/utf8/mod.rs

```diff
@@ -29,54 +29,33 @@ fn datetime_pattern<F, K>(val: &str, convert: F) -> Option<&'static str>
 where
     F: Fn(&str, &str) -> chrono::ParseResult<K>,
 {
-    [
-        // 21/12/31 12:54:98
-        "%y/%m/%d %H:%M:%S",
-        // 2021-12-31 24:58:01
-        "%Y-%m-%d %H:%M:%S",
-        // 21/12/31 24:58:01
-        "%y/%m/%d %H:%M:%S",
-        //210319 23:58:50
-        "%y%m%d %H:%M:%S",
-        // 2021/12/31 12:54:98
-        "%Y/%m/%d %H:%M:%S",
-        // 2021-12-31 24:58:01
-        "%Y-%m-%d %H:%M:%S",
-        // 2021/12/31 24:58:01
-        "%Y/%m/%d %H:%M:%S",
-        // 20210319 23:58:50
-        "%Y%m%d %H:%M:%S",
-        // note: '%F' cannot be parsed by polars native parser
-        // 2019-04-18T02:45:55
-        "%Y-%m-%dT%H:%M:%S",
-        // 2019-04-18T02:45:55[...]
-        // milliseconds
-        "%Y-%m-%d %H:%M:%S.%3f",
-        "%Y-%m-%dT%H:%M:%S.%3f",
-        // microseconds
-        "%Y-%m-%d %H:%M:%S.%6f",
-        "%Y-%m-%dT%H:%M:%S.%6f",
-        // nanoseconds
-        "%Y-%m-%d %H:%M:%S.%9f",
-        "%Y-%m-%dT%H:%M:%S.%9f",
-    ]
-    .into_iter()
-    .find(|&fmt| convert(val, fmt).is_ok())
+    let result = patterns::DATETIME_Y_M_D
```

Comment:
Is this the gist of the change? That we search in another set?

Reply:
hey - sorry, I wanted to leave some commentary but it was getting late, I should've left it in 'draft'

The gist of the change is
```diff
-                    for i in 1..(s.len() - fmt_len) {
+                    for i in 1..(s.len().saturating_sub(fmt_len)) {
```
which avoid the panic error

The change in this file is a drive-by cleanup - the formats are duplicated between here and `patterns.rs`, so this reduces the difference between `exact=True` and `exact=False`

Reply:
Great, thanks!